### PR TITLE
example: Added with-contentlayer example

### PR DESCRIPTION
**Pull request by imadatyatalah**
## Documentation / Examples

- [x] Make sure the linting passes

### CODE REVIEW

**Comment by imadatyatalah** [comment]
@leerob Can you review it?

File: examples/with-contentlayer/package.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "private": true,
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start"
+  },
+  "dependencies": {
+    "contentlayer": "^0.0.27",
+    "next": "11.1.2",
```
Comment (leerob) [inline examples/with-contentlayer/package.json:10]:
```suggestion
    "next": "latest,
```

**Review by leerob — DISMISSED** [review]

**Comment by imadatyatalah** [comment]
@leerob

**Comment by imadatyatalah** [comment]
@leerob

**Comment by imadatyatalah** [comment]
@leerob

**Comment by imadatyatalah** [comment]
@timneutkens

**Comment by imadatyatalah** [comment]
@timneutkens

**Comment by imadatyatalah** [comment]
@leerob

**Comment by imadatyatalah** [comment]
@balazsorban44 can you review my example?

**Comment by slavakurilyak** [comment]
+1 for **Contentlayer** example for MD or MDX file types

> Working with content data (e.g. Markdown files or CMS) is a surprisingly difficult and laborious task when developing modern sites (e.g. with Next.js). **Contentlayer** is a content SDK that aims to make content easy for developers with a focus on great DX and performance. ([contentlayer.dev](url))

contentlayer-02

File: examples/with-contentlayer/package.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  "private": true,
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start"
+  },
+  "dependencies": {
+    "contentlayer": "^0.1.1",
+    "next": "latest",
+    "next-contentlayer": "^0.1.1",
```
Comment (leerob) [inline examples/with-contentlayer/package.json:11]: Could we update to latest here? And React 18?

Comment (imadatyatalah) [reply examples/with-contentlayer/package.json]: @leerob done

File: examples/with-contentlayer/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+## Preview
+
+Preview the example live on [StackBlitz](http://stackblitz.com/):
+
+[![Open in StackBlitz](https://developer.stackblitz.com/img/open_in_stackblitz.svg)](https://stackblitz.com/github/vercel/next.js/tree/canary/examples/with-contentlayer)
+
+## Deploy your own
+
+Deploy the example using [Vercel](https://vercel.com?utm_source=github&utm_medium=readme&utm_campaign=next-example):
```
Comment (leerob) [inline examples/with-contentlayer/README.md:13]:
```suggestion
Deploy the example using [Vercel](https://vercel.com?utm_source=github&utm_medium=readme&utm_campaign=next-example) or preview live with [StackBlitz](https://stackblitz.com/github/vercel/next.js/tree/canary/examples/with-contentlayer)
```

File: examples/with-contentlayer/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+This example shows how to integrate Contentlayer in Next.js. Learn more about [Contentlayer](https://www.contentlayer.dev/docs).
+
+## Preview
+
+Preview the example live on [StackBlitz](http://stackblitz.com/):
+
+[![Open in StackBlitz](https://developer.stackblitz.com/img/open_in_stackblitz.svg)](https://stackblitz.com/github/vercel/next.js/tree/canary/examples/with-contentlayer)
+
+## Deploy your own
```
Comment (leerob) [inline examples/with-contentlayer/README.md:11]:
```suggestion

## Deploy your own
```

**Review by leerob — APPROVED** [review]
Thank you!

**Comment by slavakurilyak** [comment]
@imadatyatalah When deploying this example on Vercel, I receive the following errors:

```
[20:23:29.828] Cloning github.com/slavakurilyak/with-contentlayer (Branch: main, Commit: a7b5390)
[20:23:30.420] Cloning completed: 591.528ms
[20:23:30.757] Installing build runtime...
[20:23:32.692] Build runtime installed: 1.935s
[20:23:33.417] Looking up build cache...
[20:23:33.673] Build Cache not found
[20:23:33.855] Installing dependencies...
[20:23:34.187] yarn install v1.22.17
[20:23:34.208] info No lockfile found.
[20:23:34.213] [1/4] Resolving packages...
[20:23:38.386] [2/4] Fetching packages...
[20:23:52.864] [3/4] Linking dependencies...
[20:23:52.864] warning "contentlayer > @contentlayer/source-files > date-fns-tz@1.3.4" has unmet peer dependency "date-fns@>=2.0.0".
[20:23:52.867] warning "contentlayer > @contentlayer/utils > @opentelemetry/exporter-trace-otlp-grpc > @opentelemetry/core@1.0.1" has incorrect peer dependency "@opentelemetry/api@>=1.0.0 <1.1.0".
[20:23:52.867] warning "contentlayer > @contentlayer/utils > @opentelemetry/exporter-trace-otlp-grpc > @opentelemetry/resources@1.0.1" has incorrect peer dependency "@opentelemetry/api@>=1.0.0 <1.1.0".
[20:23:52.867] warning "contentlayer > @contentlayer/utils > @opentelemetry/exporter-trace-otlp-grpc > @opentelemetry/sdk-trace-base@1.0.1" has incorrect peer dependency "@opentelemetry/api@>=1.0.0 <1.1.0".
[20:23:52.879] warning Workspaces can only be enabled in private projects.
[20:23:52.880] warning Workspaces can only be enabled in private projects.
[20:23:52.884] warning Workspaces can only be enabled in private projects.
[20:24:06.302] [4/4] Building fresh packages...
[20:24:08.236] success Saved lockfile.
[20:24:08.240] Done in 34.06s.
[20:24:08.277] Detected Next.js version: 12.1.6
[20:24:08.279] Running "yarn run build"
[20:24:08.540] yarn run v1.22.17
[20:24:08.570] $ next build
[20:24:10.820] Generated 2 documents in .contentlayer
[20:24:10.842] Attention: Next.js now collects completely anonymous telemetry regarding usage.
[20:24:10.843] This information is used to shape Next.js' roadmap and prioritize features.
[20:24:10.843] You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
[20:24:10.843] https://nextjs.org/telemetry
[20:24:10.843] 
[20:24:10.882] info  - Checking validity of types...
[20:24:11.000] warn  - No ESLint configuration detected. Run next lint to begin setup
[20:24:11.003] info  - Creating an optimized production build...
[20:24:16.085] Failed to compile.
[20:24:16.085] 
[20:24:16.086] ./pages/index.js
[20:24:16.086] Module not found: Package path ./generated is not exported from package /vercel/path0/node_modules/contentlayer (see exports field in /vercel/path0/node_modules/contentlayer/package.json)
[20:24:16.086] 
[20:24:16.086] ./pages/posts/[id].js
[20:24:16.086] Module not found: Package path ./generated is not exported from package /vercel/path0/node_modules/contentlayer (see exports field in /vercel/path0/node_modules/contentlayer/package.json)
[20:24:16.086] 
[20:24:16.087] 
[20:24:16.087] > Build failed because of webpack errors
[20:24:16.123] error Command failed with exit code 1.
[20:24:16.123] info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
[20:24:16.135] Error: Command "yarn run build" exited with 1
```

**Comment by leerob** [comment]
Ah yeah looks like there's a bug - @imadatyatalah would you be willing to address that?

**Comment by imadatyatalah** [comment]
hey @slavakurilyak & @leerob, I have tried to deploy the example to Vercel and I got the same error, The error was just from the missed `jsconfig.json` file, I forget to add it sorry about that.

@slavakurilyak you can add a `jsconfig.json` file with this content and it will work.
```json
{
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "contentlayer/generated": ["./.contentlayer/generated"]
    }
  },
  "include": [".contentlayer/generated"]
}
```

@leerob Should I add `jsconfig.json` file to this PR or create a new PR to fix this error?

**Comment by leerob** [comment]
If you could make a new PR, I'm happy to get it approved and merged!

**Comment by imadatyatalah** [comment]
@leerob I have created a new PR #37257, Thanks @slavakurilyak for spotting this issue!

**Pull request merged**